sql/core/src/test/scala/org/apache/spark/sql/DatasetSuite.scala

```diff
@@ -2841,6 +2860,265 @@ class DatasetSuite extends QueryTest
     checkDataset(Seq(seqMutableSet).toDS(), seqMutableSet)
     checkDataset(Seq(mapMutableSet).toDS(), mapMutableSet)
   }
+
+  // below tests are related to SPARK-49960 and TransformingEncoder usage
+  test("Incorrect derived nullability with TransformingEncoder - non nullable") {
+    val sparkI = spark
+    type T = Tuple2[Seq[Seq[Int]], Seq[Int]]
+    val data: Seq[T] = Seq( ( Seq( Seq(1, 2, 3) ), Seq(1, 2, 3) ) )
+    // for reference only
+    val sparkDataTypeOG = {
```

Comment:
It is a bit bit distracting to add code that is not used. I understand that this represents the automated way of encoder inference.

Reply:
agreed, per above I've removed